pkg/config/cluster.go

```diff
@@ -355,6 +361,43 @@ type MetricsServer struct {
 	Disable bool `json:"disable"`
 }
 
+// OpenIDConnect feature flag
+type OpenIDConnect struct {
+	Enable bool                `json:"enable"`
+	Config OpenIDConnectConfig `json:"config"`
+}
+
+// OpenIDConnectConfig config
+type OpenIDConnectConfig struct {
+	IssuerURL      string `json:"issuer_url"`
+	ClientID       string `json:"client_id"`
+	UsernameClaim  string `json:"username_claim"`
+	UsernamePrefix string `json:"username_prefix"`
+	GroupsClaim    string `json:"groups_claim"`
+	GroupsPrefix   string `json:"groups_prefix"`
+	RequiredClaim  string `json:"required_claim"`
+	SigningAlgs    string `json:"signing_algs"`
+	CAFile         string `json:"ca_file"`
+}
+
+// Validate features config
+func (f *Features) Validate() error {
+	// Currently only validate OIDC config
+	if !f.OpenIDConnect.Enable {
+		return nil
+	}
+
+	if f.OpenIDConnect.Config.IssuerURL == "" {
+		return errors.New("oidc-issuer-url can't be empty")
+	}
+
+	if f.OpenIDConnect.Config.ClientID == "" {
+		return errors.New("oidc-client-id can't be empty")
```

Comment:
The error should mention the name of the field we use (`client_id`).

Reply:
✔️